Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3061,6 +3061,29 @@ extension BrowserViewController: PreferencesObserver {
             setupTabs()
             updateTabsBarVisibility()
             updateApplicationShortcuts()
+            
+            // Destroy all tab data
```

Comment:
The logic is good, but to save performance and to keep similar code in same place. Can you please move the clearable logic to settings itself? And do the reset configure before setup tabs